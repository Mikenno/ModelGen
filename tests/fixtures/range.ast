Module
- Call
-- Name
-- Range
--- Integer 10
--- Integer 20

- Call
-- Name
-- Range
--- Integer 10
--- Integer 20
--- Integer 2

- For
-- Name i
-- Range
--- Integer 10
--- Integer 20
-- Call
--- Name print
--- Name i

- For
-- Name i
-- Range
--- Integer 10
--- Integer 20
--- Integer 2
-- Call
--- Name print
--- Name i

- For
-- Name i
-- Range
--- Integer 10
--- Integer 20
--- Integer 2
-- Call
--- Name print
--- Name i

- Assign
-- Name x
-- Range
--- Integer 10
--- Integer 20

- Assign
-- Name y
-- Range
--- Integer 10
--- Integer 20
--- Integer 2

- Assign
-- Name z
-- Range
--- Integer 10
--- Integer 20

- Assign
-- Name w
-- Range
--- Integer 10
--- Integer 20
--- Integer 2

- For
-- Name i
-- Name x
-- Call
--- Name print
--- Name i