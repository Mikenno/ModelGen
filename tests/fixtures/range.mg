
print(10:20)
print(10:20:2)

for i in 10:20
	print(i)

for i in 10:20:2
	print(i)

for i in (10:20:2)
	print(i)

x = 10:20
y = 10:20:2
z = (10:20)
w = (10:20:2)

for i in x
	print(i)
