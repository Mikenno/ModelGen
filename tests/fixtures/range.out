[10, 11, 12, 13, 14, 15, 16, 17, 18, 19]
[10, 12, 14, 16, 18]
10
11
12
13
14
15
16
17
18
19
10
12
14
16
18
10
12
14
16
18
10
11
12
13
14
15
16
17
18
19
